include/mxnet/tuple.h

```diff
@@ -404,9 +453,11 @@ class TShape : public Tuple<dim_t> {
   }
   /*! \return total number of elements in the shape */
   inline size_t Size() const {
+    CHECK_GE(this->ndim(), 0) << "Shape is unknown.";
```

Comment:
Is this 0 or -1?

Reply:
It's CHECK_GE, so should be 0. :)

Reply:
I see. I misread just now. Suggest using `CHECK_NE` to be more specific

Reply:
I think `>= 0` is a safer check than `!= -1` to rule out the cases when ndim is mistakenly messed up ending with a negative integer other than -1.

Reply:
When it is a negative number other than -1, you should immediately throw an error, instead of silently allow this to happen.

Reply:
Makes sense. I will add this check wherever possible.